src/bench/connectblock.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2025 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <addresstype.h>
+#include <bench/bench.h>
+#include <interfaces/chain.h>
+#include <kernel/cs_main.h>
+#include <script/interpreter.h>
+#include <sync.h>
+#include <test/util/setup_common.h>
+#include <validation.h>
+
+#include <cassert>
+#include <vector>
+
+CBlock CreateTestBlock(
+    std::shared_ptr<TestChain100Setup> test_setup,
+    const std::vector<CKey>& keys,
+    const std::vector<CScript>& scriptpubkeys,
+    int num_txs = 1000)
+{
+    Chainstate& chainstate{test_setup->m_node.chainman->ActiveChainstate()};
+
+    std::vector<CTxOut> outputs;
+    // Each transaction will pay to the same outputs that will be spent in the next one
+    outputs.reserve(scriptpubkeys.size());
+    for (const auto& spk : scriptpubkeys) {
+        outputs.emplace_back(COIN, spk);
+    }
+
+    const WitnessV1Taproot coinbase_taproot{XOnlyPubKey(test_setup->coinbaseKey.GetPubKey())};
+
+    // Create the outputs that will be spent in the first transaction of the test block
+    // Doing this in a separate blocks excludes the validation of it's inputs from the benchmark
+    auto coinbase_to_spend{test_setup->m_coinbase_txns[0]};
+    const auto [first_tx, _]{test_setup->CreateValidTransaction(
+        std::vector{coinbase_to_spend},
+        std::vector{COutPoint(coinbase_to_spend->GetHash(), 0)},
+        chainstate.m_chain.Height() + 1, keys, outputs, std::nullopt, std::nullopt)};
+    const auto test_block_parent_coinbase{GetScriptForDestination(coinbase_taproot)};
+    test_setup->CreateAndProcessBlock(std::vector{first_tx}, test_block_parent_coinbase, &chainstate);
+
+    std::vector<CMutableTransaction> txs;
+    txs.reserve(num_txs);
+    CTransactionRef input_tx{MakeTransactionRef(first_tx)};
+    for (int i{0}; i < num_txs; i++) {
+        std::vector<COutPoint> inputs;
+        inputs.reserve(outputs.size());
+
+        for (size_t j{0}; j < outputs.size(); j++) {
+            inputs.emplace_back(input_tx->GetHash(), j);
+        }
+
+        const auto [taproot_tx, _]{test_setup->CreateValidTransaction(
+            std::vector{input_tx}, inputs, chainstate.m_chain.Height() + 1, keys, outputs, std::nullopt, std::nullopt)};
+        txs.emplace_back(taproot_tx);
+        input_tx = MakeTransactionRef(taproot_tx);
+    }
+
+    // Coinbase output can use any output type as it is not spent and will not change the benchmark
+    const CScript coinbase_spk{GetScriptForDestination(coinbase_taproot)};
+    return test_setup->CreateBlock(txs, coinbase_spk, chainstate);
+}
+
+std::pair<std::vector<CKey>, std::vector<CScript>> CreateKeysAndScripts(const CKey& coinbaseKey, size_t num_taproot, size_t num_nontaproot)
```

Comment:
I know that we generically call things "scripts" but I think it would be more clear to name this something like `CreateKeysAndSignatures`, as that better indicates what we are testing, i.e., ECDSA signatures and Schnorr signatures. I'd also recommend renaming the variables to `num_ecdsa_sigs` and `num_schnorr_sigs`, or something similar.